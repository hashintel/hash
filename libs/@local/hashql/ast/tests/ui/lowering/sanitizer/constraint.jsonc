//@ run: fail
//@ description: Generic constraints with bounds should be sanitized and error out
[
  "add<T: Number>",
  //~^ ERROR Remove this constraint from 'T'
  { "#literal": 1 },
  { "#literal": 2 }
]
