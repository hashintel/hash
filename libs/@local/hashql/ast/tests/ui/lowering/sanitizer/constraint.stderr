[31m[sanitizer::invalid-generic-constraint] Error:[0m Invalid generic constraint
   â•­â”€[ <unknown>:4:8 ]
   â”‚
 4 â”‚   "add<T: Number>",
   â”‚        â”€â”€â”€â”€â”¬â”€â”€â”€â”€  
   â”‚            â•°â”€â”€â”€â”€â”€â”€ Remove this constraint from 'T'
   â”‚ 
   â”‚ Help: Generic constraints (like 'T: Bound') are not allowed in this context. Use just the parameter name without bounds: 'T'. For example, change 'T: Clone' to just 'T'.
   â”‚ 
   â”‚ Note: Generic constraints with bounds can only be used in certain positions such as function declarations, type declarations and newtype declarations. In other contexts, like usage sites, only the parameter name should be specified.
â”€â”€â”€â•¯
